client/my-sites/github-deployments/deployments/index.tsx

```diff
@@ -38,8 +38,8 @@ export function GitHubDeployments() {
 
 		return (
 			<>
-				<HeaderCake isCompact>
-					<h1>{ __( 'Connect repository' ) }</h1>
+				<HeaderCake isCompact onClick={ () => page( '/sites' ) } backIcon="chevron-left">
```

Comment:
I think we just need to update the styles and the back link keeps hidden if there is no connected repo.